src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/MirrorEndpointsSpec.groovy

```diff
@@ -234,44 +232,39 @@ class MirrorEndpointsSpec extends HealthCheckSpecification {
                 .withAtLeastNTraffgensOnSource(2)
                 .withAtLeastNNonOverlappingPaths(2)
                 .random()
-        def flow = flowHelperV2.randomFlow(swPair).tap { it.allocateProtectedPath = true }
-        flowHelperV2.addFlow(flow)
+        def flow = flowFactory.getBuilder(swPair)
+                .withProtectedPath(true)
+                .build().create()
 
         when: "Create a mirror point"
-        def mirrorTg = swPair.src.traffGens[1]
-        def mirrorEndpoint = FlowMirrorPointPayload.builder()
-                .mirrorPointId(flowHelperV2.generateFlowId())
-                .mirrorPointDirection(mirrorDirection.toString().toLowerCase())
-                .mirrorPointSwitchId(swPair.src.dpId)
-                .sinkEndpoint(FlowEndpointV2.builder().switchId(swPair.src.dpId).portNumber(mirrorTg.switchPort)
-                        .vlanId(flowHelperV2.randomVlan())
-                        .build())
-                .build()
-        flowHelperV2.createMirrorPoint(flow.flowId, mirrorEndpoint)
+        def mirrorTg = swPair.src.traffGens[1] // take TG at the same switch but not used in the flow
```

Comment:
swPair.src.traffGens.find { it.switchPort != flow.source.portNumber }

Reply:
Sorry, I missed this line, fixed